### DISCUSSION TITLE
Not optimized for large amounts of data

### DISCUSSION THREAD

**Discussion by huzzi** [insideapp-srl/flutter_supabase_chat_core#14] [Q&A]
Hello 👋 

Thank you so much for lovely package, You mentioned this package is not optimized for large amounts of data. What is the concern? Maybe someone can investigate this and create a PR? Thanks again.

 _Recommended for small or PoC projects, might not be optimized for large amounts of data. I suggest to use this on a free plan, otherwise be extremely cautious._

**Comment by rickypid** [comment]
Hi @huzzi , 

thanks for your interest. 

It was simply written as I've never used it (I haven't received any feedback from users on GitHub) on a large project although it is certainly scalable. 

You have to respect the limitations of your Supabase priano regarding `real-time` https://supabase.com/docs/guides/realtime/quotas. A few indexes also need to be optimised, but I am working on this and they will be added in the next release. Let me know how you plan to use this project or how you have used it.

**Reply by huzzi** [reply]
Hi @rickypid 

Thank you for your reply, I will do some testing with large number of messages and will let you know the feedback.

I'm looking forward to the next release. 

Thank you.

**Comment by rickypid** [comment]
@huzzi ,

After your question I started to delve into the scalability issue, I improved a few points by streaming all users or all appearance rooms of a user by converting them into paged api calls, this to avoid huge data flows. 

In addition, I implemented missing indexes and performed some security fixes on the RLS. I believe that with this PR https://github.com/insideapp-srl/flutter_supabase_chat_core/pull/16 we should reach a good level of quality. However, I still maintain that some tests on large tables will be necessary to confirm the scalability of the functionality of this package.

**Reply by huzzi** [reply]
@rickypid Appreciate your time and effort you're putting into this, I'm planning to use this chat on an app I'm working on that requires just one-to-one chat feature. Once the PR is merged I'll play around and give you some feedback. So far it's looking good 👍👍

Thanks again

**Comment by rickypid** [comment]
@huzzi,

https://github.com/insideapp-srl/flutter_supabase_chat_core/releases/tag/v1.3.1
